### DISCUSSION TITLE
need help about this mithril.js code...could someone please advise

### DISCUSSION THREAD

**Discussion by kangaroolab** [MithrilJS/mithril.js#2901] [General]
saw below as part of the lifecycle example but couldn't tell what exactly it is doing...is it to create vnode for javascript functions? appreciate any advice.

                subnodes.map((key, i) =>
                    m(tag, {
                        key,
                        remove: () =>
                            subnodes.splice(i, 1),
                    })
                ),

**Comment by cavemansspa** [comment]
hello @kangaroolab. that snippet looks a bit odd, where are you seeing this example? the lifecycle methods start with `on` (i.e. defining an `onremove` hook will result in that function being be called when a dom element is removed from the document)

**Comment by kangaroolab** [comment]
here you go...

https://flems.io/#0=N4IgtglgJlA2CmIBcBmArAOgAwBYA0IAZhAgM7IDaoAdgIZiJIgYAWALmLCAQMYD21NvEHIQIAL54a9RswBW5XgKEimhAK7UebCAIAEAGT4BzABQBKPcAA61Pfb0I2e-pqEAnPQF49ARlsOjvDOsCbeehQAurYBDk56DOE2doEOtDCmDKSktMbwlsmpRY4mGAAO6qQspsDQSC58bvDuANQteHo6DPXU8ADuegAitEIWHVk5efUTufDi5rHFDu7B6u52DIupkjEpRaFmBVtL-NSkfAgYB6YHC3vbeLtLYLRlpoRH90srbGt2BxgXm8PscHOItuDqFsfn8EvBbJDbAA3WieA7hXoDIyHWwAelxAPSUFMAHIhKRnL4SeYMETSeTnAAmal4gmlOlk+AUvQoFlQ6gAQTKZSSWyREH6pHqwEhiMFwow4slGGx4Q0Wh0AgsVieSziwT0UD4YHCmig8GIvSgur1QWciR8yVBtqVfXqpiR1D45ssXgAfDqvrbUvijSaAGThvSkYIAFQgDEabFMK2o5vcd2Deph6wizqzqTApIARuo2GwBCSOoUC7XAgIeLAIDwANbu30Bmt1uuQ7tZnZBvupEkAJXgUHctD6JPz3fMj35Q6KRZJpfLlers6X9a0Tdb7vy3k7W+3RXgGBWE6natosBjJ9PgXRPkxhhMFgfj-sqfTH8HX-sXsALBBdgIcEkAGEEFRGd-znUCvxXNcK2oKtAzA+wIGoXp3AACVjABZAx6hJAAhFZaBbKtPzrBs9zbPRTEPf10IwwJz0vScBh8Qhb3vOCv0JDISQAKjEvk2MCH9mj-NigMfeYFxo20VzQrtgIbCihHdT1vWY48BMfMNwl080MDDZS+2k9xZIwgc2IEdQyigEZ4HbI89Gs2yFPnG0+2iQztgfOUsxzDZ4UXYN1W0XQ7C8z5uzAC9hF-MMOgoSzAiLawQFIMpaChbg8xACC1lTZwujcnKOlfYZRhpCsAGU2HcLDDkiXzAqWAEgVMUwIDYDtWMfMKElJMoVjQgApRqAHkADkMApVrqGMCBCAAT36wbMyXeZLI6vylnkoLqHxQJIXxOUhTKRUJT6UgMHmvS1U0GL9A9L0fUDLZ4hbeANpMr7zxGFrHv+jbfoNUh1GLYHSHCKI-NG9SdywgadOBobUeKITiQhvQWj0Ek9AAWlJvQBHRtgJOKAd800+BXMxvTsYfPHTAJomSfJymtC0+BaaKen-0c5zmcY0z9OGvUOa54myYpsWXKEIWHiOhwBGLC0+BWJyVbcyWsY8nH9nZDJ5Z5pXqG1whdfgfXXLVi6lNFm2dZWFYwD4JFDc+1mTfZ838YBwmFd5rWPeaeBvd952QI1+wBC9n2-altnArl0PucVvmU7j3bhddpZXXdU2szYXIEL7UH3ClKx87mavuyTCo2Gb4N5kDyLbSzwGc9511GIpVFBr5dmDR+AA1YHwhXCgKQ2hAvGLXX03qXwygAD2jC5oAAbj0fKYDa+oMDQGPD5edw1uoM+L7ASI1MkPMuvA57zWot-7AhjvgxJr+llG56EjGNVcZYUJqUyg4LCOF8JERIlvL+246LNgYo3P+tpFKWRXCwdw1Ji51hhnDPSj1eqcwBh0CAHZoH2CLJXYwm5v56l-rQ1Ijd3L+jYUsYh8MlplD3PAfqHRfCdW3PtZhnUcElggRuGW3ZYHNHgcRYmLRkFLlQfuRiNDmE8NhnwioVQKH92JhgEmRNTC8NIVcYQxg2AsDDqIwuBZsGBUOj3PU+IYxsHjImMsfUM5Zj7mHK2egh5MTTOYeORQrp3AfKNaewN8wnTlHKJK3s3CmCNDwdQDBBAYFXlADaTDAium1OXUaGVArz0XsvQgCAt71CwH4PQaAsAAFJD6p3cPUvgfRSYbXqLQMsfB95P3GKYG6d1lQf3gNWCGG95hiODDUtgS94BeHqfARpLTfCtI6V032PTQj9MGXoYZFYxloSLFM10j1sTmGcYEAKF0ER3G4LleACB3pnFEDgAAnEgNpEgpAgDoAwUQGAeDZA+acFQbBRCFMBjWKAEA8qwFoGcrZW996xGxaTVFKwfn1H4LAXJ1BcUCVXlvUmpAIAAC9T56FXu4dMpNqWUqWKi9FmL6gOGxZy4odtBB8tSBgAAHDHaMBVSC0uaOtQVRQyh8DpZqO+-KIBb3HIq1IfRoD2NFQ4XwWADn5hYPACAxh2CGr8Cazp+Zr63xtVgHVgRj6otWoa8+l98wVjKDahwLr8wIEIGwANegg2iyOb0t0DgWDQHNBShEuwJrwHkcUR1WFnUYEZA-Sl4gPkxm+Wq8gTBfAAHYkBYAkJEAgTZqAtlLVQMFMhRCQHsa1LgBA1hcCYOwNgZQpT4k0GUFsxgoXGlxO2vBJAAACvgMALoAGxToGjO2AGAFAfLWWUWQpAeCtTKAi8QkRxBAA

I think I figure it out..the tag is simply App.views.Node itself. What the code does is to create a new vnode for the child node...

**Comment by cavemansspa** [comment]
that's correct. the `m(tag` at line 142 is effectively the same as: `m(App.views.Node`.
